ui/library/src/components/thumbnails/Thumbnail.tsx

```diff
@@ -0,0 +1,63 @@
+import classnames from 'classnames';
+import * as React from 'react';
+
+import { PageRenderContext } from '../../context/PageRenderContext';
+import { ScrollContext } from '../../context/ScrollContext';
+import { getMaxVisibleElement } from '../../utils/MaxVisibleElement';
+import { Nullable } from '../types/utils';
+
+type Props = {
+  pageNumber: number;
+};
+
+export const Thumbnail: React.FunctionComponent<Props> = ({ pageNumber }: Props) => {
+  const { pageRenderStates, buildObjectURLForPage, getObjectURLForPage } =
+    React.useContext(PageRenderContext);
+  const { isPageVisible, scrollToPage, visiblePageRatios } = React.useContext(ScrollContext);
+
+  const [maxVisiblePageNumber, setMaxVisiblePageNumber] = React.useState<Nullable<string>>(null);
+
+  const objectURL = getObjectURLForPage({ pageNumber });
+
+  React.useEffect(() => {
+    if (visiblePageRatios.size !== 0) {
+      const max = getMaxVisibleElement(visiblePageRatios);
+      if (max) {
+        setMaxVisiblePageNumber(max.toString());
+      }
+    }
+  }, [visiblePageRatios]);
+
+  const hasPageVisible =
```

Comment:
This name is highly overlapped with `isPageVisible`. It would be hard to tell their difference simply by looking at them. Consider a better name to increase the readability.